cli/dcoscli/experimental/main.py

```diff
@@ -71,17 +72,22 @@ def _info():
     return 0
 
 
-def _add(dcos_package):
+def _add(dcos_package, package_name, package_version):
     """
     Adds a DC/OS package to DC/OS
 
     :param dcos_package: path to the DC/OS package
-    :type dcos_package: str
+    :type dcos_package: None | str
+    :param package_name: the name of a remote DC/OS package
+    :type package_name: None | str
+    :param package_version: the version of a remote DC/OS package
+    :type package_version: None | str
     :return: process status
     :rtype: int
     """
     package_manager = get_package_manager()
-    response = package_manager.package_add(dcos_package)
+    response = package_manager.package_add(
+        dcos_package, package_name, package_version)
```

Comment:
What do you think about doing some processing of the command-line arguments here, before calling methods on `package_manager`? For example, you could ensure that `dcos_package` and `package_name` are not both specified.

The reason for doing that is to decouple the `package_manager` interface from the command-line arguments, allowing the interface to be more direct. In fact, we should probably split `package_manager.package_add()` into two methods that each add from a single source.